### DISCUSSION TITLE
only one node is porcessing request in cluster, no load balance

### DISCUSSION THREAD

**Discussion by xiaojuan8** [milvus-io/milvus#15946] [Q&A and General discussion]
Hi, I deployed a milvus cluster, which has 4 nodes in the cluster.
But when I call search from the client with parallel requesting, but I found only one node is processing request, no load balance for cluster worked.
I have 4 nodes:
10.30.88.57
10.30.88.60
10.30.88.58
10.30.88.59

When I request cluster with search() call, only first node above is on high CPU and memory utility, the other three node is low utility on CPU and memory(3%CPU). I wish every node is working for my search request.

my client code :

milvus::ConnectParam connect_param{"10.30.88.57", 32491};
auto status = client->Connect(connect_param);
CheckStatus("Failed to connect milvus server:", status);
const auto progress_monitor = ::milvus::ProgressMonitor::NoWait();
status = client->LoadCollection(collection_name, progress_monitor);
multi-thread block
{
... 
          milvus::ConnectParam connect_param{"10.30.88.57", 32491};
          auto status = client_t->Connect(connect_param);
          milvus::SearchArguments arguments{};
          arguments.SetCollectionName(collection_name);
          arguments.AddPartitionName(partition_name);
          arguments.SetTopK(TOP_K);
          arguments.SetMetricType(COLLECTION_METRIC_TYPE);
          arguments.AddOutputField(field_mid_name);
          arguments.AddExtraParams("nprobe", std::to_string(nprobe));
          // set to strong guarantee so that the search is executed after the inserted data is persisted
          arguments.SetGuaranteeTimestamp(milvus::GuaranteeStrongTs());
          for (int64_t i = 0; i < NQ; i++) {
            srand((unsigned)time(NULL));
            int index = rand() % (mid_list.size() - NQ) + i;
            arguments.AddTargetVector(field_embedding_name, emb_list[index]);
          }
           milvus::SearchResults search_results{};
          status = client_t->Search(arguments, search_results);
...
}

the nodes is deployed as shown below:
![image](https://user-images.githubusercontent.com/74544828/157364555-657d7383-f0eb-453f-a0d4-7dc5bfa49620.png)

**Comment by xiaojuan8** [comment]
I tried again. I found that no matter which node IP I connect, only first node (10.30.88.57) is high on CPU and memory, the other  three is not consuming CPU resource.

**Comment by yhmo** [comment]
How many rows in the collection? If there is only one segment in the collection, it cannot be balanced(only one query node is busy). If there are only two segments in the collection, you will see only two query nodes are busy during searching.

Use the get_query_segment_info() to show how many segments are there:
```
from pymilvus import utility

infos = utility.get_query_segment_info(collection_name=collection_name)
print(infos)
```

**Reply by xiaojuan8** [reply]
OK👌, I have tried that my segment shows size = 3, But I want every node in my cluster work for my search request. What should I do to make segment more or equal to my cluster node num? @yhmo

**Reply by engrabdu** [reply]
Hello @xiaojuan8 i want to deploy cluster i have 3 vms ..can you please share me the steps to b follow  or any materials thanks

**Reply by yhmo** [reply]
> OKok_hand, I have tried that my segment shows size = 3, But I want every node in my cluster work for my search request. What should I do to make segment more or equal to my cluster node num? @yhmo

The **shard_num** parameter can affect segment count. It is a parameter of create_collection(). The default value is 2, whcih means the collection use two data channel to consume inserted data. This parameter only can be defined when creating a collection. You can create a collection with shard_num=4, and observe the segment count.
